Mono.Debugging.Soft/SoftDebuggerSession.cs

```diff
@@ -1969,27 +1970,30 @@ void HandleBreakEventSet (Event[] es, bool dequeuing)
 					}
 				}
 			}
-			
+
+			if (currentRequest == null || (currentRequest != null && es[0] != null && es[0].Request != null && es[0].Request.getId () != currentRequest.getId ()))
+				currentRequest = es[0].Request;
+
+
+			if (!redoCurrentStep && currentRequest != null) {
+				currentRequest.Enabled = false;
+				currentRequest = null;
+			}
+
+
 			if (redoCurrentStep) {
-				StepDepth depth = currentStepRequest.Depth;
-				StepSize size = currentStepRequest.Size;
-				
+				StepDepth depth = (currentRequest as StepEventRequest).Depth;
+				StepSize size = (currentRequest as StepEventRequest).Size;
+
```

Comment:
These should probably be regular casts instead of as-casts, unless they can be a request type other than StepEventRequest, in which case this code would need to handle the as-cast resulting in null.

Reply:
Done